pallets/elections/src/lib.rs

```diff
@@ -133,14 +151,17 @@ pub mod pallet {
                 _ if on_chain == StorageVersion::new(1) => {
                     migrations::v1_to_v2::pre_upgrade::<T, Self>()
                 }
+                _ if on_chain == StorageVersion::new(2) => {
+                    migrations::v2_to_v3::pre_upgrade::<T, Self>()
+                }
                 _ => Err("Bad storage version"),
             }
         }
         #[cfg(feature = "try-runtime")]
         fn post_upgrade() -> Result<(), &'static str> {
             let on_chain = <Pallet<T> as GetStorageVersion>::on_chain_storage_version();
             match on_chain {
-                _ if on_chain == STORAGE_VERSION => migrations::v1_to_v2::post_upgrade::<T, Self>(),
+                _ if on_chain == STORAGE_VERSION => migrations::v2_to_v3::post_upgrade::<T, Self>(),
```

Comment:
Shouldn't we somehow match it so that we can migrate from any version to the latest? I've seen it somewhere in Polkadot I think

Reply:
We should and it is done in the fn `on_runtime_upgrade` line 116. This is fn for performing checks after the upgrade already happened.

Reply:
We should and it is done in the fn `on_runtime_upgrade` line 116. This is fn for performing checks after the upgrade already happened.

Reply:
We should and it is done in the fn `on_runtime_upgrade` line 116. This is fn for performing checks after the upgrade already happened.